internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -89,19 +90,100 @@ Overview of the connection process
 	MsSQL->Driver: Login Response
 	Driver->Secretless: Login success or failure
 	Secretless->Client: Login Response (Premade)
- */
+*/
 
 // SingleUseConnector is used to create an authenticated connection to an MSSQL target
 type SingleUseConnector struct {
-	backendConn net.Conn
-	clientConn  net.Conn
-	logger      log.Logger
+	backendConn       net.Conn
+	clientConn        net.Conn
+	logger            log.Logger
+	newMSSQLConnector types.NewMSSQLConnectorFunc
```

Comment:
Ah, ok this one is a little different from the previous one above, where we both agreed that `MSSQLConnectorCtor` is a better name for the _type_.  

But here we're talking about an instance of a `MSSQLConnectorCtor`, that is, a constructor function itself.  For those we have a very clear convention in Go with the `NewXXX`.

Consider how it looks when we use it:

```go
driverConnector, err := connector.newMSSQLConnector(dataSourceName(connDetails))
```

That reads exactly like normal code where we do `foo := NewFoo(blah)`.  But if we where to name the instance using ctor, we'd get:

```go
driverConnector, err := connector.mssqlConnectorCtor(dataSourceName(connDetails))
```
which makes it sound as if we're _returning_ a ctor, rather than invoking one to return a connector instance.  It also breaks the standard `NewXXX` convention.

So in this case I vote to keep the name as is -- wdyt?

And as a side note, I don't "contructor" in the name violates the rule which prohibits including the _data type_ in a name.  "constructor" is an abstract description telling you what the thing is.  The rule is more about names like `FirstNameStr` and so on.  

Arguably the `Func` suffix should be avoided though for the reason you state.

Reply:
Ah, yes, I absolutely agree. The sample invocation that you provided above makes it abundantly clear.